### PR TITLE
[SYCL][ESIMD][Doc] Add DPAS example

### PR DESCRIPTION
**Pull request by turinevgeny**

### CODE REVIEW

File: sycl/doc/extensions/supported/sycl_ext_intel_esimd/examples/README.md [hunk 1]
```diff
@@ -65,4 +65,102 @@ is to show the basic ESIMD APIs in well known examples.
...
-3) TODO: Add more examples here.
+3) Dot Product Accumulate Systolic (DPAS) API - ["dpas"](./dpas.md)
+
+   Please see the full source code here: ["dpas"](./dpas.md)
+
+   ```c++
+    // Res = A * B.
+    // Assume the HW is PVC.
+
+    constexpr int SystolicDepth = 8;
```
Comment (sarnex) [inline sycl/doc/extensions/supported/sycl_ext_intel_esimd/examples/README.md:76]: nit: i think we are currently doing constexpr vars or defines in `CAPS_AND_UNDERSCORES` code style, but maybe that's only for globals constexprs? not a big deal either way

File: sycl/doc/extensions/supported/sycl_ext_intel_esimd/examples/README.md [hunk 2]
```diff
@@ -65,4 +65,102 @@ is to show the basic ESIMD APIs in well known examples.
...
+    constexpr int BElemBitSize = 4; // 4-bit integers.
+
+    // Elements of A and B will are packed into uint8_t,
+    // meaning that one uint8_t holds two 4-bit unsigned integers.
+    // Packaging for A and res is horizontal, for B is vertical.
+    using PackedType = unsigned char;
+    using APackedType = PackedType;
+    using BPackedType = PackedType;
+
+    // res type, according to documentation is either int or uint.
```
Comment (sarnex) [inline sycl/doc/extensions/supported/sycl_ext_intel_esimd/examples/README.md:94]:
```suggestion
    // result type, according to documentation is either int or uint.
```

just making it a little more clear, it took me a quick second to figure it out

**Review by sarnex — APPROVED** [review]
only a couple nits, i didn't review the algorithm in-depth, i assume that's from an existing test which is correct

File: sycl/doc/extensions/supported/sycl_ext_intel_esimd/examples/dpas.md [hunk 3]
```diff
@@ -0,0 +1,268 @@
...
+    for (int i = 0; i < K; i++) {
+      for (int j = 0; j < N; j++) {
+        int value = (i + j % 4) == 0 ? 1 : (2 + i + j) % 3;
+        write_to_vertically_packed_matrix_b(b_packed, i, j,
+                                            static_cast<BPackedType>(value));
+        assert(value == (int)(static_cast<BPackedType>(value)) && "ERROR");
+      }
+    }
+
+    q.single_task([=]() SYCL_ESIMD_KERNEL {
```
Comment (sarnex) [inline sycl/doc/extensions/supported/sycl_ext_intel_esimd/examples/dpas.md:226]: In the only in-tree example of an ESMID kernel we are using `[[intel::sycl_explicit_simd]]`, so maybe we should use that for consistency?

File: sycl/doc/extensions/supported/sycl_ext_intel_esimd/examples/README.md (hunk 1 above)
Comment (turinevgeny) [reply sycl/doc/extensions/supported/sycl_ext_intel_esimd/examples/README.md:76]: It looks like it is mostly `CapitalizedWords` in esimd headers.

File: sycl/doc/extensions/supported/sycl_ext_intel_esimd/examples/README.md (hunk 2 above)
Comment (turinevgeny) [reply sycl/doc/extensions/supported/sycl_ext_intel_esimd/examples/README.md:94]: Initially I was referring to variable `res`, but it is much better just to say `Result`.

File: sycl/doc/extensions/supported/sycl_ext_intel_esimd/examples/dpas.md (hunk 3 above)
Comment (turinevgeny) [reply sycl/doc/extensions/supported/sycl_ext_intel_esimd/examples/dpas.md]: Yes, makes sense.

**Comment by turinevgeny** [comment]
> only a couple nits, i didn't review the algorithm in-depth, i assume that's from an existing test which is correct

Thank you for review, Nick!

**Review by sarnex — APPROVED** [review]

**Comment by turinevgeny** [comment]
@intel/llvm-gatekeepers, the patch is ready for merging.

**Pull request merged**